cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/hvdc/LossFactor.java

```diff
@@ -0,0 +1,132 @@
+
+import java.util.Objects;
+
+import com.powsybl.cgmes.conversion.Context;
+import com.powsybl.iidm.network.HvdcLine;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+
+class LossFactor {
+
+    LossFactor(Context context, HvdcLine.ConvertersMode mode, double pAC1, double pAC2, double poleLossP1,
+        double poleLossP2) {
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(mode);
+        this.context = context;
+        this.mode = mode;
+        this.pAC1 = pAC1;
+        this.pAC2 = pAC2;
+        this.poleLossP1 = poleLossP1;
+        this.poleLossP2 = poleLossP2;
+        this.lossFactor1 = Double.NaN;
+        this.lossFactor2 = Double.NaN;
+    }
+
+    void compute() {
+        // compute loss factors
+
+        if (pAC1 != 0 && pAC2 != 0) {
+
+            // we only keep one as we are not sure if pAC1 and pAC2 are consistent
+            if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER)) { // we ignore pAC2
+                computeLossFactor1(pAC1, poleLossP1, mode);
+                computeLossFactor2FromPAC1(pAC1, poleLossP1, poleLossP2, mode);
+            } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER)) { // we ignore pAC1
+                computeLossFactor1FromPAC2(pAC2, poleLossP1, poleLossP2, mode);
+                computeLossFactor2(pAC2, poleLossP2, mode);
+            }
+        } else if (pAC1 != 0) { // pAC2 == 0
+
+            computeLossFactor1(pAC1, poleLossP1, mode);
+            computeLossFactor2FromPAC1(pAC1, poleLossP1, poleLossP2, mode);
+        } else if (pAC2 != 0) { // pAC1 == 0
+
+            computeLossFactor1FromPAC2(pAC2, poleLossP1, poleLossP2, mode);
+            computeLossFactor2(pAC2, poleLossP2, mode);
+        } else {
+            this.lossFactor1 = 0.0;
+            this.lossFactor2 = 0.0;
+        }
+
+        if (Double.isNaN(this.lossFactor1)) {
+            this.lossFactor1 = 0.0;
+            context.fixed("lossFactor1", "was NaN", Double.NaN, this.lossFactor1);
+        }
+        if (Double.isNaN(this.lossFactor2)) {
+            this.lossFactor2 = 0.0;
+            context.fixed("lossFactor2", "was NaN", Double.NaN, this.lossFactor2);
+        }
+
+        // else (i.e. pAC1 == 0 && pAC2 == 0) do nothing: loss factors are null and
+        // stations are probably disconnected
+    }
+
+    private void computeLossFactor1(double pAC1, double poleLossP1, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER)) { // pAC1 > 0
+            this.lossFactor1 = (poleLossP1 / pAC1) * 100;
+        } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && Math.abs(pAC1) + poleLossP1 != 0) { // pAC1 < 0
+            this.lossFactor1 = (poleLossP1 / (Math.abs(pAC1) + poleLossP1)) * 100;
```

Comment:
You have to remove the multiplying by 100. We made a mistake before because the documentation is wrong. LossFactor is a ratio.

Reply:
Done